### PR TITLE
Added ability to specify 'role' field for select speaker messages for Group Chats (addresses #1861)

### PR DESCRIPTION
**Pull request by marklysze**
## Why are these changes needed?
As per feature request #1861 ("[Feature Request]: Allowing user to specify the "role" field for select speaker messages"), this PR includes changes to the GroupChat constructor to allow the user to specify the role for the select speaker messages. This addresses the issue of role names when using Mistral through Mistral.AI's API.

Changes:
- Added `role_for_select_speaker_messages` to `GroupChat` constructor in `groupchat.py`
- Added @ekzhu's Mistral.AI notebook to documentation under Non-OpenAI models section
- Added @ekzhu's updates to the Using Non-OpenAI about page
- Added a Tips for Non-OpenAI models page to documentation (should be a basis for future tips)
- Added a sub-section on the Conversation Patterns tutorial explaining the new attribute and provided an example

## Related issue number
Addresses #1861 

## Checks

- [X] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [X] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/agentchat/conversable_agent.py [hunk 1]
```diff
@@ -716,7 +716,7 @@ def _print_received_message(self, message: Union[Dict, str], sender: Agent):
             if "function_call" in message and message["function_call"]:
                 function_call = dict(message["function_call"])
                 func_print = (
-                    f"***** Suggested function call: {function_call.get('name', '(No function name found)')} *****"
+                    f"***** Suggested function Call: {function_call.get('name', '(No function name found)')} *****"
```
Comment (ekzhu) [inline autogen/agentchat/conversable_agent.py:719]:
```suggestion
                    f"***** Suggested function call: {function_call.get('name', '(No function name found)')} *****"
```

File: autogen/agentchat/conversable_agent.py [hunk 2]
```diff
@@ -728,7 +728,7 @@ def _print_received_message(self, message: Union[Dict, str], sender: Agent):
                 iostream.print(colored("*" * len(func_print), "green"), flush=True)
             if "tool_calls" in message and message["tool_calls"]:
                 for tool_call in message["tool_calls"]:
-                    id = tool_call.get("id", "No tool call id found")
+                    id = tool_call.get("id", "(No id found)")
```
Comment (ekzhu) [inline autogen/agentchat/conversable_agent.py:731]:
```suggestion
                    id = tool_call.get("id", "No tool_call_id found")
```

File: autogen/agentchat/conversable_agent.py [hunk 3]
```diff
@@ -1311,12 +1311,6 @@ def _generate_oai_reply_from_client(self, llm_client, messages, cache) -> Union[
                 )
             for tool_call in extracted_response.get("tool_calls") or []:
                 tool_call["function"]["name"] = self._normalize_name(tool_call["function"]["name"])
-                # Remove id and type if they are not present.
```
Comment (ekzhu) [inline autogen/agentchat/conversable_agent.py:1314]: These were for Mistral AI compatibility. Are these not needed now?

**Comment by marklysze** [comment]
Closing this PR as it was based on an older version of main.

New PR #2199

**Pull request closed without merging**